keras_nlp/tokenizers/byte_pair_tokenizer_test.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_nlp.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+
+
+class BytePairTokenizerTest(tf.test.TestCase):
```

Comment:
Let's add a little more coverage, for some key use cases

 - A batched tf dataset
 - A unbatched tf dataset
 - A function annotated with @tf.function
 - Some more complex unicode character cases (maybe we can validate these first with the original tokenizer impl to make sure we have it right)